### ISSUE TITLE
Add Support for Phrase Matching

### ISSUE THREAD

**Issue by kevinfodness** [alleyinteractive/elasticsearch-extensions#29]
# User Story

> As a user, I want to be able to enter a phrase in quotes and only see results that match that phrase, e.g., "cool beans" should only match "cool" followed by "beans" rather than each word individually (as is the current default).

# Acceptance Criteria

- Add a new method to the controller called `enable_phrase_matching` that, when called as part of configuration, causes phrase matching to be turned on for all queries that are run through the search implementation.

# Implementation Details

Example of how this could be used in code:

```php
add_action(
	'elasticsearch_extensions_config',
	 function( $es_config ) {
		$es_config->enable_phrase_matching();
	 }
);
```

- When enabled, search queries should be evaluated for text in double quotes, which should be included in the search as a `match_phrase`.
- Any other text in the search, not in quotes, should be searched for normally.
- The solution should support multiple phrase searches in the same query (e.g., `"quick brown" fox "jumped over"`).
- This feature should be opt-in and should be disabled by default.
- There should be a corresponding `disable_phrase_matching` function to turn this feature off, which could be used by plugins or themes conditionally to unset this setting in particular contexts (otherwise there would be no way to turn it off once it was turned on).

**Comment by mboynes** [comment]
We have prior art for this, reach out about that before beginning work on this.

**Comment by jakewrfoster** [comment]
closed by #67
